### ISSUE TITLE
2 issues with inner join -- issue 1

### ISSUE THREAD

**Issue by chris3392** [sqlitebrowser/sqlitebrowser#2440]
Details for the issue
--------------------
doing an inner join executing current statement only through appropriatemenu button, will not stop execution at the comma and show results; instead will continue and execute next statement and hence show result of said immediately following statement.

#### What did you do?

Select Account.name, account.personemail from Account inner join ListNBRDoublon on ListNBRDoublon.personemail = account.personemail  ; ;

select 3*5 ;

#### What did you expect to see?
result of inner join statement ie a table with Account.name, account.personemail

(that table appeared briefly until replaced by result of immiately following statement as was executed)

#### What did you see instead?

result of 3*5, ie, 15

Useful extra information
-------------------------
> DB4S v3.12.0 [built for x86_64-little_endian-lp64] on Ubuntu 18.04.5 LTS (linux/5.4.0-48-generic) [x86_64]
> using SQLCipher Version 3.4.1 (based on SQLite 3.30.1)
> and Qt 5.9.5

**Comment by chrisjlocke** [comment]
Technically this is a duplicate of #2429.
If you only highlight the instructions you want to run, this helps matters, so highlight a line, run it, then highlight the next line.

**Comment by chrisjlocke** [comment]
Closing as this is a duplictate, but fee;l free to comment if needed. 😃

**Comment by chris3392** [comment]
Not sure if this is a duplicate or not. I have seen that behavior only with the inner join instruction. Evertything else is fine

As for issue #2, which I am hinting i will post - but have not-- I found in the meantime it was irrelevant as an Sqlite limitation.

Tx

**Comment by chrisjlocke** [comment]
Its a duplicate, as you're doing two queries, but the results window only shows the last query.  While a bug, its easily worked around by only executing one instruction at a time, via selective highlighting.  Its not ideal, but works fairly well once you get used to it.  I tend to use one SQL tab with multiple queries so I can just select the one I want and run it.

**Comment by chris3392** [comment]
Unfortunately I cannot reproduce it anymore as i changed my database too heavily.

If it helps, i was not using selective highlighting but was requesting a one instruction only execution  by using the relevant button from db browser.

I think it worth to mention, since I have the occasion, i am using that tool very often and really like it. Could hardly live without it.
Congratulation for the work.

**Comment by chrisjlocke** [comment]
>  i was not using selective highlighting

Oh no, that's the issue - that's the bug (acknowledged in the other issue).  If you were, it would have worked out OK. 😃 
Hopefully someone will work out how to fix it, as its a common problem - especially if you use MS SQL Management Studio.

**Comment by chris3392** [comment]
ok, 
Not intending to start trolling here.

hard for me to understand how to view these as same error. DBBrowser is executing 2 queries where only the last one output is displayed, while I explicitely hit the button to execute ONE query, not two (and two were executed instead)

Furthermore, said bug is mentioning only ONE query, which output does not show the rows properly but the columns are shown  properly.

While in my case, the output (both rows and column names) is correct for  execution of query #2; the problem being, why on earth does it execute query #2 after #1, while being requested to execute query #1 only.

Sounds very different (though I can understand it could result from same technical bug, but outputs and execution are just at odds)

**Comment by chrisjlocke** [comment]
> while I explicitely hit the button to execute ONE query

Apologies - so you pressed the 'Execute current line' button?

![image](https://user-images.githubusercontent.com/7872218/96011689-24e85f80-0e3b-11eb-871d-b26afd1c9cd4.png)

We've had many 'quirks' with this, as the parser can get confused with different lines depending on the OS (Linux vs Windows ... carriage returns vs carriage returns with line-feeds)
This works on Windows, but if you are pressing the 'execute current line', then yes, that's an error.  Apologies - while you did mention 'the appropriate menu', I didn't appreciate that's the button you were referring to.

**Comment by chris3392** [comment]
No worry.
Correct. I was pressing that execute current line button.
Ubuntu 18.04 here, if it helps.

**Comment by chrisjlocke** [comment]
OK, if you put the text carat on line 1 and press that button, do you get the result of line 2?
Just making sure 100% the error is the error I'm reading.... :)

**Comment by chris3392** [comment]
Yes, this is what I meant.
I put the text carat on line 1 and press that button,
then I can see the columns appearing for a very brief period of time (that query retruns a few thousands rows) and i have no time to see the rows appearing once they are returned
then, the columns and the rows of the next query appear. And it stops there.

So, query 1 was executed(expected)  then immediately query 2 is executed as well (unexpected).
the result of query 2 is on screen, which is expected when it executes; the unexpected is query 2 to execute at all in that scenario.

**Comment by chrisjlocke** [comment]
Excellent, thanks for clarifying. :)
